### ISSUE TITLE
[java] Private method with interfaces as parameters are marked as unused, even though they are used

### ISSUE THREAD

**Issue by Cunning0fDesires** [pmd/pmd#4581]
**Affects PMD Version:** 

6.55.0

**Description:**

A private method within a class accepts an interface as a parameter, but specific implementations are being passed when the method is called. When passing the pmd check, it fails saying that I have an unused private method, even though it's used.

This works well when I add @SuppressWarnings({"PMD.UnusedPrivateMethod"}) to the methods which accept interfaces, so that the warning is ignored

**Exception Stacktrace:**

42024 [INFO] PMD Failure: Rule:UnusedPrivateMethod Priority:3 Avoid unused private methods such as 'someMethod()'..

**Steps to reproduce:**

1.  Create an interface with some methods to be implemented
2. Create a class which implements this interface (let's call it A)
3. Create another class (B) with one public method and one private. The private method should accept the interface as a parameter.
4. Call the private method in the public method in the class B, but passing the class A as a parameter in place of the interface

**Running PMD through:** Maven

**Comment by jsotuyod** [comment]
If I understood correctly, the following code should trigger the issue:

```java
public class Foo {
	public interface Marker {}
	
	public class MyMarker implements Marker {}
	
	private void doMarker(Marker b) {
		// This is called
	}
    
    public void doIt() {
    	doMarker(new MyMarker());
    }
}
```

Under PMD 7.0.0-rc3 this is not the case. The rule handles this as expected.
